sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -180,9 +180,13 @@ class Dataset[T] private[sql](
     // to happen right away to let these side effects take place eagerly.
     queryExecution.analyzed match {
       case c: Command =>
-        LocalRelation(c.output, queryExecution.executedPlan.executeCollect())
```

Comment:
Yeah, the check I added to ensure we get the same results in the SQL tab has [several hundred failures](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/75579/testReport/) that go through this. Looks like the path is almost always `spark.sql` when the SQL statement is a command like CTAS.

I like your version and will update.